ci/check-crates.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# input:
+# 	env:
+#  	 	- CRATE_TOKEN
+#   	- COMMIT_RANGE
+
+if [[ -z $COMMIT_RANGE ]]; then
+  echo "COMMIT_RANGE should be provided"
+  exit 1
+fi
+
+if ! command -v toml &>/dev/null; then
+  echo "not found toml-cli"
+  cargo install toml-cli
+fi
+
+declare skip_patterns=(
+  "Cargo.toml"
+  "programs/sbf"
+)
+
+declare -A verified_crate_owners=(
+  ["solana-grimes"]=1
+)
+
+# get Cargo.toml from git diff
+readarray -t files <<<"$(git diff "$COMMIT_RANGE" --diff-filter=AM --name-only | grep Cargo.toml)"
+printf "%s\n" "${files[@]}"
+
+error_count=0
+for file in "${files[@]}"; do
+  read -r crate_name package_publish workspace < <(toml get "$file" . | jq -r '(.package.name | tostring)+" "+(.package.publish | tostring)+" "+(.workspace | tostring)')
+  echo "=== $crate_name ($file) ==="
+
+  if [[ $package_publish = 'false' ]]; then
+    echo -e "⏩ skip (package_publish: $package_publish)\n"
+    continue
+  fi
+
+  if [[ "$workspace" != "null" ]]; then
+    echo -e "⏩ skip (is a workspace root)\n"
+    continue
+  fi
+
+  for skip_pattern in "${skip_patterns[@]}"; do
+    if [[ $file =~ ^$skip_pattern ]]; then
+      echo -e "⏩ skip (match skip patterns)\n"
+      continue 2
+    fi
+  done
+
+  result="$(cargo owner --list -q "$crate_name" --token "$CRATE_TOKEN" 2>&1)"
+  if [[ $result =~ ^error ]]; then
+    if [[ $result == *"Not Found"* ]]; then
+      ((error_count++))
+      echo "❌ new crate $crate_name not found on crates.io. you can either
+
+1. mark it as not for publication with 'package.publish = false' in its Cargo.toml
```

Comment:
let's be a little more explicit here.  spare us questions on discord :wink: 

```suggestion
1. mark it as not for publication in its Cargo.toml

    [package]
    ...
    publish = false
```

Reply:
https://github.com/solana-labs/solana/pull/29009/commits/259eada2fdb6a1dc26ab9a9e1f3cff87c4a20456